notebooks_community/clf_constrained_bo.ipynb

```diff
@@ -0,0 +1,840 @@
+      "text/plain": [
+       "<Figure size 500x400 with 2 Axes>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "# Plot the townsend function and constraint\n",
+    "import matplotlib.pyplot as plt\n",
+    "import numpy as np\n",
+    "\n",
+    "def plot_townsend(ax):\n",
+    "    x = np.linspace(-2.5, 2.5, 100)\n",
+    "    y = np.linspace(-2.5, 2.5, 100)\n",
+    "    X, Y = np.meshgrid(x, y)\n",
+    "\n",
+    "    obj = townsend(torch.tensor(np.stack([X, Y], axis=-1), **tkwargs)).cpu().numpy()\n",
+    "    con = townsend.constraint(torch.tensor(np.stack([X, Y], axis=-1), **tkwargs)).cpu().numpy()\n",
+    "\n",
+    "    #mask out the constraint region<0\n",
+    "    obj[con==0] = np.nan\n",
+    "\n",
+    "    c = ax.contourf(X, Y, obj, levels=20,cmap='Blues')\n",
+    "\n",
+    "    ax.set_xlabel(\"X1\")\n",
+    "    ax.set_ylabel(\"X2\")\n",
+    "    ax.set_title(\"Townsend Problem\")\n",
+    "\n",
+    "    plt.colorbar(c, ax=ax, orientation='vertical')\n",
+    "    return ax\n",
+    "\n",
+    "# Plot the townsend function and constraint\n",
+    "fig, ax = plt.subplots(1, 1, figsize=(5, 4))\n",
+    "plot_townsend(ax)\n",
+    "plt.tight_layout()\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Generate Training Data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[KeOps] Warning : \n",
+      "    The default C++ compiler could not be found on your system.\n",
+      "    You need to either define the CXX environment variable or a symlink to the g++ command.\n",
+      "    For example if g++-8 is the command you can do\n",
+      "      import os\n",
+      "      os.environ['CXX'] = 'g++-8'\n",
+      "    \n",
+      "[KeOps] Warning : Cuda libraries were not detected on the system or could not be loaded ; using cpu only mode\n"
+     ]
```

Comment:
Is there way to get rid of this warning as this is completely unrelated to the tutorial?